### ISSUE TITLE
Why the max length of paragraph is 300?

### ISSUE THREAD

**Issue by shuxiaobo** [unilight/R-NET-in-Tensorflow#8]
Hi, @unilight 
I have run the code successfully, but when i review the code, I found the max length of the paragraph is 300, as I know , the dataset of SQuAD max length of paragraph is much bigger than 300 word, could you help to figure this problem? thanks

**Comment by unilight** [comment]
Hi @shuxiaobo,

The main reason why I empirically chose to truncate all training paragraphs to 300 words was that the memory was no enough. If my GPU was equipped with larger memory, I would love to extend this number to a larger one.

Nonetheless, using all training data doesn't always lead to better performance. Due to tensorflow API design, all training paragraphs need to be padded to the same length, i.e. the short paragraphs will be padded to be as long as the longest paragraph. When doing back propagation through time, this might cause gradient issues, such as gradient vanishment. In RNN training, there are certain techniques to deal with this problem. For example, in mini-batch training, one can sort the length of the training paragraphs and put those paragraphs with similar length in the same batch, so they don't need to be padded to fit the longest paragraph. However, at the time I implemented R-Net, I wasn't really sure how to do this. Together with the memory reason, I decided to just truncate all paragraphs to a specific length.

Let me know if you have further questions!

**Comment by shuxiaobo** [comment]
Thanks. you're right. this model need a large  gpu memory. the model I imple used almost all three P80 GPUs. but also got a bed result. there is so many trick in the model. but the model you implemented has got the best result is also a good new, I learned a lot from your code. thanks a lot.

**Comment by unilight** [comment]
You're welcome. 👍

**Comment by shuxiaobo** [comment]
Hi Mr @unilight :
    Could you please help me to figure out when should I add a dropout layer in the training process? because I trained my model and BiDAF(self imple) model . I found the train loss have a great difference. like:
![image](https://user-images.githubusercontent.com/11539889/42418539-3835ece0-82d5-11e8-929f-ec6a944f814e.png)
I have dropout everytime I use the rnn.
